### PR TITLE
Dataset based fallout configurations

### LINKED ISSUE #446: Create dataset based fallout configurations
Create dataset based fallout configurations.

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Has fallout configuration that runs test based on data from a file.
**Which issue(s) this PR fixes**:
Fixes #446 

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: fallout/sample-dataset.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{"hash":"3576be49f989ac551af8f7e7b2aad0d4f9dd7b3882e7843839aa173d017cf7fe","ver":2,"vin_sz":1,"vout_sz":1,"size":220,"weight":880,"fee":6468,"relayed_by":"0.0.0.0","lock_time":0,"tx_index":8970844506661621,"double_spend":false,"time":1685054037,"block_index":null,"block_height":null,"inputs":[{"sequence":4294967295,"witness":"","script":"4730440220771152f9eac8397baf82af8711969bdffb4b84db116c3cdf90c15d02699dcbd30220527f2d88855566337933a2180db8ee5455c3da2558f2f79b37dbeb283fd0bbf4014104e62bff7b1af952b1bfd6b68af9e6367d69756807b69f8920a762e1b500d86d821e8d11d3bf9a8bcd88d7a97b6a6aaa945be8e15bd1111478f37c919f807be61f","index":0,"prev_out":{"addr":"1koboutKx17hEXb9mDrF4YjULkdE5faxr","n":0,"script":"76a9140848c944ba84c10a6cb754910a2e8311f49c00b888ac","spending_outpoints":[{"n":0,"tx_index":8970844506661621}],"spent":true,"tx_index":3507365639401646,"type":0,"value":384372}}],"out":[{"type":0,"spent":false,"value":377904,"spending_outpoints":[],"n":0,"tx_index":8970844506661621,"script":"001499c3c085b9b7e31909b4a8ae61a1acc8d63df647","addr":"bc1qn8puppdekl33jzd54zhxrgdvertrmaj8j36jsu"}]}
```
Comment (maheshrajamani) [inline fallout/sample-dataset.txt:1]: Should we need to push big dataset file? I think this is good enough don't want to make the repo heavy.

Comment (ivansenic) [inline fallout/sample-dataset.txt:1]: we already have this in the `nosqlbench/` folder, why duplicating? would remove

File: fallout/dataset-test-nosqlbench.yaml [hunk 2]
```diff
@@ -0,0 +1,297 @@
+min_version: "5.17.3"
```
Comment (ivansenic) [inline fallout/dataset-test-nosqlbench.yaml:1]: what was the reason for adding a new file, why not using the one from the `nosqlbench/`? Where do they differ?

Comment (ivansenic) [reply fallout/dataset-test-nosqlbench.yaml:1]: Aha I see now that this is not only the CRUD.. I was thinking we agreed dataset tests will use more crud-like approach..

File: fallout/dataset-test-nosqlbench.yaml [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+  sort_key: WeightedStrings('<<sort_key:data.size>>'); ToString();
+
+  # http request id
+  request_id: ToHashedUUID(); ToString();
+
+  # autogenerate auth token to use on API calls using configured uri/uid/password, unless one is provided
+  token: Discard(); Token('<<auth_token:>>','<<uri:http://localhost:8081/v1/auth>>', '<<uid:cassandra>>', '<<pswd:cassandra>>');
+
+  seq_key: Mod(<<docscount:10000>>); ToString() -> String
+  update_seq_key: Mod(<<docscount:10000>>); ToString() -> String
```
Comment (ivansenic) [inline fallout/dataset-test-nosqlbench.yaml:42]: seems not needed, just use `seq_key`

File: fallout/dataset-test-nosqlbench.yaml [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        Accept: "application/json"
+        X-Cassandra-Request-Id: "{request_id}"
+        X-Cassandra-Token: "{token}"
+        Content-Type: "application/json"
+        ok-body: ".*\"data\".*"
+        body: >2
+          {
+            "find" : {
+              "filter" : {
+                "full_name": { "$exists": true }
```
Comment (ivansenic) [inline fallout/dataset-test-nosqlbench.yaml:195]: should this also use `exists_key`?

File: fallout/dataset-test-fallout.yaml [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        module: nosqlbench
+        properties:
+          alias: main_{{op}}
+          service_type: stargate
+          cycles: {{op_cycles}}
+          args:
+            - driver=http
+            - workload=https://raw.githubusercontent.com/stargate/jsonapi/{{test_branch}}/fallout/dataset-test-nosqlbench.yaml
+            - tags=block:main,name:.*{{op}}
+            - threads=1x
```
Comment (ivansenic) [inline fallout/dataset-test-fallout.yaml:130]: why threads `1x`? wouldn't this be too low?

File: fallout/dataset-test-fallout.yaml [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+# field name to be used to check the $exists filter
+exists_key: data
+
+# load setup:
+# - total of 1M ops split between 10 operations equally
+# - each operation limited to 25ops/s, totals to 250ops/s
+# - each operation will be executed by 3 clients, totals to 30 clients
+op_cycles: 100000
+op_rate: 25
+op_connections: 3
```
Comment (ivansenic) [inline fallout/dataset-test-fallout.yaml:37]: these were smoke test setup, do we want to have the same ones here as well?

**Review by ivansenic — APPROVED** [review]
Seems good, I will approve, but can you point me to the successful Fallout run please?

File: fallout/dataset-test-nosqlbench.yaml [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+              },
+              "projection": {
+                "full_name": 1,
+                "data": 1
+              }
+            }
+          }
+
+      # aka find with filter and sort
+      find-multi-find-by-group-sort-by-sort-key:
```
Comment (ivansenic) [inline fallout/dataset-test-nosqlbench.yaml:205]: maybe names should be actually adapted to the action done?

Comment (maheshrajamani) [reply fallout/dataset-test-nosqlbench.yaml:205]: Will change.

File: fallout/dataset-test-nosqlbench.yaml (hunk 4 above)
Comment (maheshrajamani) [reply fallout/dataset-test-nosqlbench.yaml:195]: Will change it.

File: fallout/dataset-test-nosqlbench.yaml (hunk 2 above)
Comment (maheshrajamani) [reply fallout/dataset-test-nosqlbench.yaml:1]: Didn't understand this statement. Can we connect on call?

File: fallout/sample-dataset.txt (hunk 1 above)
Comment (maheshrajamani) [reply fallout/sample-dataset.txt:1]: Thought we keeping the fallout resources separate from nosqlbench.

**Comment by maheshrajamani** [comment]
> Seems good, I will approve, but can you point me to the successful Fallout run please?

http://fallout.aws.dsinternal.org/tests/ui/mahesh.rajamani@datastax.com/dataset-test/c63ef5b5-667b-4d51-8341-367ffebe30da/artifacts

**Pull request merged**